tests/puppeteer/pages/Install/install.js

```diff
@@ -129,33 +129,35 @@ module.exports = class Install extends CommonPage {
   /**
    * Check if database exist (if not, it will be created)
    * and check if all set properly to submit form
+   * @return {Promise<boolean>}
    */
-  async checkDatabaseConnected() {
+  async isDatabaseConnected() {
     await this.page.click(this.testDbConnectionButton);
     // Create database 'prestashop' if not exist
     if (await this.elementVisible(this.createDbButton, 3000)) {
       await this.page.click(this.createDbButton);
     }
-    await this.page.waitForSelector(this.dbResultCheckOkBlock, {visible: true});
+    return this.elementVisible(this.dbResultCheckOkBlock, 3000);
   }
 
   /**
    * Check if prestashop is installed properly
    */
-  async checkInstallationSuccessful() {
-    await this.page.waitForSelector(this.installationProgressBar, {visible: true});
-    await this.page.waitFor(this.installationProgressBar, {visible: true});
-    await this.page.waitFor(this.generateSettingsFileStep, {visible: true});
-    await this.page.waitFor(this.installDatabaseStep, {visible: true});
-    await this.page.waitFor(this.installDefaultDataStep, {visible: true, timeout: 360000});
-    await this.page.waitFor(this.populateDatabaseStep, {visible: true, timeout: 360000});
-    await this.page.waitFor(this.configureShopStep, {visible: true, timeout: 360000});
-    await this.page.waitFor(this.installModulesStep, {visible: true});
-    await this.page.waitFor(this.installModulesAddons, {visible: true, timeout: 360000});
-    await this.page.waitFor(this.installThemeStep, {visible: true});
-    await this.page.waitFor(this.installFixturesStep, {visible: true});
-    await this.page.waitForSelector(this.finalStepPageTitle, {visible: true, timeout: 90000});
-    await this.checkStepTitle(this.finalStepPageTitle, this.finalStepEnTitle);
+  async isInstallationSuccessful() {
+    await Promise.all([
+      this.page.waitForSelector(this.installationProgressBar, {visible: true}),
+      this.page.waitForSelector(this.generateSettingsFileStep, {visible: true}),
+      this.page.waitForSelector(this.installDatabaseStep, {visible: true, timeout: 60000}),
+      this.page.waitForSelector(this.installDefaultDataStep, {visible: true, timeout: 120000}),
+      this.page.waitForSelector(this.populateDatabaseStep, {visible: true, timeout: 180000}),
+      this.page.waitForSelector(this.configureShopStep, {visible: true, timeout: 240000}),
+      this.page.waitForSelector(this.installModulesStep, {visible: true, timeout: 360000}),
+      this.page.waitForSelector(this.installModulesAddons, {visible: true, timeout: 360000}),
+      this.page.waitForSelector(this.installThemeStep, {visible: true, timeout: 360000}),
+      this.page.waitForSelector(this.installFixturesStep, {visible: true, timeout: 360000}),
+      this.page.waitForSelector(this.finalStepPageTitle, {visible: true, timeout: 360000}),
+    ]);
+    return this.checkStepTitle(this.finalStepPageTitle, this.finalStepEnTitle);
```

Comment:
Oof that's a big Promise you got there...

Reply:
yep :smile: